Add methods to extract PLHIV and population data

r-ash opened the pull request:
At the request of Roma Bhatkoti at CDC can we add methods to get population and PLHIV data from Spectrum. This PR will add those 2 functions `extract_population` and `extract_hiv_population`. I've also added some notes about installation and usage to the readme. 

So can we review both that the code is up to scratch and that you think the notes and documentation are sufficient that anyone trying to use this will be able to use these 2 new functions.

r-ash commented:
Also I wanted to request Jeff to review this too but I can't work out how I can add him. I'm guessing the list of possible reviewers is managed by some group membership. Possible at the mrc-ide level? Do you know how I can request him as a reviewer for PRs in this repo? Google had no obvious clear answer.

richfitz commented:
I created a `specio` group with jeff (and the 2 of us in it) and gave it read permissions on the repo, which means @jeffeaton can now be requested for review.

It's a bit weird because I would have thought you might want anyone to be able to review things really

richfitz commented on NAMESPACE:
```diff
@@ -1,5 +1,7 @@
 # Generated by roxygen2: do not edit by hand
 
+export(extract_hiv_population)
```
I don't love these names - before we make the package we should consider the strategy for designing the API

r-ash replied:
Agreed there not amazing, I wonder if you have any good suggestions? At least for these two new functions. If they are going to be used by others it would be nice to get something good now as to not change them in the future. 

For this name I tried to use their language i.e. the request was "we need to extract PLHIV and population"

richfitz commented on README.md:
```diff
@@ -4,4 +4,15 @@
+
+## Usage
+
+As well as the functions required for EPP `specio` also provides 2 more general functions for extracting population data and PLHIV data from the PJNZ file. These are `extract_population` and `extract_hiv_population`. See [website reference](https://mrc-ide.github.io/specio/reference/index.html) for usage and examples.
```
Can you open a ticket for a usage vignette? (You already have the one on the internal format)

r-ash replied:
Created that now. https://vimc.myjetbrains.com/youtrack/issue/mrc-273

richfitz commented on README.md:
````diff
@@ -4,4 +4,15 @@
+## Installation
+
+```r
+# install.packages("devtools")
````
Let chat about this vs the drat at some point...

r-ash commented:
I've bumped the version number and added a `NEWS.md` file. 

It would be good to get some functions names here which we are happy with if people are going to be using them. @jeffeaton do you have any input on function naming?

r-ash commented:
Following is Jeff's points on this PR

* Can we have an option to return in 'long' format
```
type.convert(as.data.frame.table(population, responseName = "totpop"))
type.convert(as.data.frame.table(hivpop, responseName = "hivpop"))
```
* Can we add similar function for ART population
* For function naming Jeff prefers things like `get_totpop()`, `get_hivpop()` and `get_artpop()`

richfitz commented:
Re: `get_hivpop` - use your judgement on names, but I personally find that if I make abbrevations in names I always end up looking it up later because it's never done completely consistently, or ending up with an abbreviation collision. I would really suggest a `_` between `hiv` and `pop` though in any case...

r-ash commented:
I'm not a fan of (most) abbreviated names either. I have a similar experience to you in that I forget what the abbreviations referred to. I feel like there are exceptions to that though. Where the word appears frequently and the abbreviation is not ambiguous with another frequently used word in the domain. I think `population` to `pop` is one of the cases. 

I've updated the names of the methods to `get_total_pop`, `get_hiv_pop` and `get_art_pop` which I think hits Jeff's requirements but is still readable.

I've added the function to extract art data and also to return in long format.

jeffeaton commented:
Thanks. What do you think about `get_*()` versus `read_*()`.  Elsewhere I’ve used `read_*()` to get specific items out of the DP file, but that was not a careful decision and I don’t feel strongly other than desire to consolidate around one convention.  We could also add  `_dp` or `_pjnz` for clarity versus the myriad other get/read things in R. E.g. options:

`get_totpop()`
`read_totpop()`
`get_pjnz_totpop()`
`read_pjnz_totpop()`
`get_dp_totpop()`
`read_dp_totpop()`

Thanks,
Jeff


On 23 May 2019, at 10:42, Rob <notifications@github.com<mailto:notifications@github.com>> wrote:


I'm not a fan of (most) abbreviated names either. I have a similar experience to you in that I forget what the abbreviations referred to. I feel like there are exceptions to that though. Where the word appears frequently and the abbreviation is not ambiguous with another frequently used word in the domain. I think population to pop is one of the cases.

I've updated the names of the methods to get_total_pop, get_hiv_pop and get_art_pop which I think hits Jeff's requirements but is still readable.

I've added the function to extract art data and also to return in long format.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/mrc-ide/specio/pull/8?email_source=notifications&email_token=ABYVOO5R6NW4DSCNLQ654LLPWZRHNA5CNFSM4HNUUCT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWBVKYY#issuecomment-495146339>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ABYVOO6LJK63WFRM5F2PCBDPWZRHNANCNFSM4HNUUCTQ>.
